jboss/container/wildfly/s2i/2.0/bash/artifacts/opt/jboss/container/wildfly/s2i/maven-s2i-overrides

```diff
@@ -14,9 +14,167 @@ function maven_s2i_custom_binary_build() {
+      else
+        log_error "Not a server home directory, exiting."
+        exit 1
+      fi
      fi
   fi
   chmod -R ug+rwX $JBOSS_HOME
 } 
+
+function create_managed_server_from_metadata() {
+    managed_server_war_file="${1}"
+
+    # Unzip the config files
+    mkdir /tmp/managed-server-files
+    unzip_file_to_directory "${managed_server_war_file}" "server-config.xml" "/tmp/managed-server-files"
+    unzip_file_to_directory "${managed_server_war_file}" "server-init.cli" "/tmp/managed-server-files"
+    unzip_file_to_directory "${managed_server_war_file}" "server-init.yml" "/tmp/managed-server-files"
+
+    # Get the layers from server-config.xml and set GALLEON_PROVISION_LAYERS
+    # If the datasource galleon layer is needed, the function will recorded it in the '$found_datasource_layers' env var
+    read_layers_from_server_config_xml "/tmp/managed-server-files/server-config.xml"
+
+    # Set the required GALLEON_PROVISION_FEATURE_PACKS env var
+    set_galleon_provision_feature_packs_env_var "${found_datasource_layers}"
+
+    # Copy the war into /deployments - the provisioning will take care of copying it for us
+    cp "${managed_server_war_file}" /deployments/ROOT.war
+
+    log_info "Provisioning server with feature packs: ${GALLEON_PROVISION_FEATURE_PACKS} - and layers: ${GALLEON_PROVISION_LAYERS}"
+    galleon_provision_server
+
+
+    add_cli_script_and_yml_if_present "/tmp/managed-server-files" "server-init.yml" "server-init.cli"
+}
+
+function unzip_file_to_directory() {
+    zip_file="${1}"
+    search_file="${2}"
+    output_dir="${3}"
+
+    # Check both META-INF/ and WEB-INF/classes/META-INF to give user some flexibility. Error if found in both places
+    # For the '&& found_xxxx=1 || found_xxxx=0' end part of each command, the '1' is hit if the file could be extracted,
+    # and the '0' if not
+
+    unzip -p "${zip_file}" "META-INF/${search_file}" >> "${output_dir}/${search_file}" && found_meta_inf=1 || found_meta_inf=0
+    unzip -p "${zip_file}" "WEB-INF/classes/META-INF/${search_file}" >> "${output_dir}/${search_file}" && found_web_inf=1 || found_web_inf=0
+
+    if [ $found_meta_inf -eq 1 ] && [ found_web_inf -eq 0 ]; then
+      log_error "Found ${search_file} in both META-INF/ and in WEB-INF/classes/META-INF/ of ${zip_file}. It should only be in one of the locations."
+      exit 1
+    fi
+}
+
+function set_galleon_provision_feature_packs_env_var() {
+    found_datasource_layers="${1}"
+
+    validate_env_var_set "MANAGED_SERVER_SERVER_FEATURE_PACK"
+    validate_env_var_set "MANAGED_SERVER_CLOUD_FEATURE_PACK_GA"
+    validate_env_var_set "MANAGED_SERVER_CLOUD_FEATURE_PACK_VERSION"
+    validate_env_var_set "MANAGED_SERVER_DATASOURCES_FEATURE_PACK_GA"
+    validate_env_var_set "MANAGED_SERVER_DATASOURCES_FEATURE_PACK_VERSION"
+
+
+    GALLEON_PROVISION_FEATURE_PACKS="${MANAGED_SERVER_SERVER_FEATURE_PACK},${MANAGED_SERVER_CLOUD_FEATURE_PACK_GA}:${MANAGED_SERVER_CLOUD_FEATURE_PACK_VERSION}"
```

Comment:
Yo ushould perhaps check first place, as soon as you detect that this is a managed war that GALLEON_PROVISION_LAYERS and GALLEON_PROVISION_FEATURE_PACKS are not set.